ios/FluentUI/BottomSheet/BottomSheetViewController.swift

```diff
@@ -0,0 +1,404 @@
+                currentBottomSheetViewController.view.accessibilityViewIsModal = futureState == .expand
+                currentBottomSheetViewController.updateResizingHandleViewAccessibility()
+                UIAccessibility.post(notification: .layoutChanged, argument: currentBottomSheetViewController.resizingHandleView)
+
+                completion?()
+                if futureState == .expand {
+                    currentBottomSheetViewController.delegate?.bottomSheetViewControllerDidExpand?(currentBottomSheetViewController)
+                } else {
+                    currentBottomSheetViewController.removeDimmingView()
+                    currentBottomSheetViewController.delegate?.bottomSheetViewControllerDidCollapse?(currentBottomSheetViewController)
+                }
+            }
+        }
+    }
+
+// MARK: - pan gesture utilities
+
+    @objc private func handlePan(gesture: UIPanGestureRecognizer) {
+        let positionInSuperview = gesture.translation(in: view.superview)
+
+        switch gesture.state {
+        case .began:
+            panningStart()
+        case .changed:
+            panningProgress(in: positionInSuperview.y)
+        case .ended:
+            let velocity = gesture.velocity(in: view)
+            panningStop(in: positionInSuperview.y, velocity: velocity.y)
+        default:
+           break
+        }
+    }
+
+    private func panningStart() {
+        if animator?.isRunning ?? false {
+           return
+        }
+
+        setupAnimator()
+    }
+
+    private func panningStop(in verticalPos: CGFloat, velocity: CGFloat) {
+        panGestureRecognizer.isEnabled = false
+
+        let windowHeight = view.window?.frame.height ?? 0
+        let shouldReverse: Bool
+        let futureState: BottomSheetViewState
+        switch currentState {
+        case .expand:
+            if verticalPos >= windowHeight / 2 || velocity > Constants.velocityThreshold {
+                shouldReverse = false
+                futureState = .collapse
+            } else {
+                shouldReverse = true
+                futureState = .expand
+            }
+        case .collapse:
+            if verticalPos <= -windowHeight / 2 || velocity <= -1 * Constants.velocityThreshold {
+                shouldReverse = false
+                futureState = .expand
+            } else {
+                shouldReverse = true
+                futureState = .collapse
+            }
+        }
+
+        animator?.isReversed = shouldReverse
+        addAnimatorCompletion(to: futureState, completion: nil)
+        animator?.continueAnimation(withTimingParameters: nil, durationFactor: 0)
+    }
+
+    private func panningProgress(in verticalPos: CGFloat) {
+        if animator?.isRunning ?? false {
+           return
+        }
+
+        let windowHeight = view.window?.frame.height ?? 0
+        var progress = -1 * verticalPos / (windowHeight - collapsedHeight)
```

Comment:
This means that when the preferredExpandedHeight is lower than the full screen, the pan gesture no longer maps 1:1 to the finger moving on the screen and it feels kinda weird. Generally sheets will move with the finger 1:1, regardless of where the sheet is going to stop. Once the sheet reaches the expanded height, we could add resistance where the pan no longer moves it 1:1, and on release it bounces back to the max height. We could also reuse that for the resistance bounce in the opposite direction that rdeassis suggested.